hangupsbot/plugins/telesync/__init__.py

```diff
@@ -698,13 +743,128 @@ def tg_command_get_me(bot, chat_id, args):
+
+        applicable_configurations = []
+        ho2tg_dict = self.configuration["memory"]["ho2tg"]
+        if conv_id in ho2tg_dict:
+            # combine config.json and memory options to generate a dict
+            config_clone = dict(self.configuration["config"])
+            config_clone.update({ self.configkey: [ ho2tg_dict[conv_id] ],
+                                  "hangouts": [ conv_id ] })
+            applicable_configurations.append({ "trigger": conv_id,
+                                               "config.json": config_clone })
+
+        return applicable_configurations
+
+    @asyncio.coroutine
+    def _send_to_external_chat(self, config, event):
+        conv_id = config["trigger"]
+        external_ids = config["config.json"][self.configkey]
+
+        user = event.passthru["original_request"]["user"]
+        message = event.passthru["original_request"]["message"]
+
+        """migrated from telesync _on_hangouts_message():
 
-    telesync_config = bot.config.get_by_path(['telesync'])
-    if not telesync_config['enabled']:
+        * by this point of execution, applicable_configuration() would have
+            already filtered only relevant events
+        * telesync configuration only allows 1-to-1 telegram-ho mappings, this
+            migrated function supports multiple telegram groups anyway"""
+
+        sync_text = message
+        photo_url = None
+
+        has_photo, photo_file_name = yield from _telesync_is_valid_image_link(sync_text)
+        if has_photo:
+            photo_url = sync_text
+            sync_text = "shared an image"
+
+        bridge_user = self._get_user_details(user, { "event": event })
+        username = bridge_user["preferred_name"]
+        if bridge_user["chat_id"]:
+            username = "<a href=\"https://plus.google.com/u/0/{}/about\">{}</a>".format( bridge_user["chat_id"],
+                                                                                         bridge_user["preferred_name"] )
+
+        chat_title = format(self.bot.conversations.get_name(conv_id))
+
+        if "chatbridge" in event.passthru and event.passthru["chatbridge"]["source_title"]:
+            chat_title = event.passthru["chatbridge"]["source_title"]
+
+        if "sync_chat_titles" not in config or config["sync_chat_titles"] and chat_title:
+            formatted_text = "{} ({}): {}".format( username,
+                                                   chat_title,
+                                                   sync_text )
+        else:
+            formatted_text = "{}: {}".format( username,
+                                              sync_text )
+
+        # send messages first
+        for eid in external_ids:
+            yield from tg_bot.sendMessage( eid,
+                                           formatted_text,
+                                           parse_mode = 'HTML',
+                                           disable_web_page_preview = True )
```

Comment:
Messages from slackrtm -> hangouts -> telesync with a newline do not sync

eg:
```
testing

message
```
will successfully send from slackrtm -> hangouts, but will not sync from hangouts -> telesync

Error message:
```
2017-03-22 08:41:38 ERROR handlers: sending: plugins.telesync._broadcast : coroutine
Traceback (most recent call last):
  File "/usr/local/share/hangoutsbot/hangupsbot/handlers.py", line 340, in run_pluggable_omnibus
    yield from function(*_passed)
  File "/usr/local/share/hangoutsbot/hangupsbot/webbridge/__init__.py", line 151, in _broadcast
    passthru = passthru ))
  File "/usr/local/share/hangoutsbot/hangupsbot/plugins/telesync/__init__.py", line 831, in _send_to_external_chat
    disable_web_page_preview = True )
  File "/usr/local/share/hangoutsbot/venv/lib/python3.5/site-packages/telepot/aio/__init__.py", line 81, in sendMessage
    return await self._api_request('sendMessage', _rectify(p))
  File "/usr/local/share/hangoutsbot/venv/lib/python3.5/site-packages/telepot/aio/__init__.py", line 70, in _api_request
    return await api.request((self._token, method, params, files), **kwargs)
  File "/usr/local/share/hangoutsbot/venv/lib/python3.5/site-packages/telepot/aio/api.py", line 111, in request
    return await _parse(r)
  File "/usr/local/share/hangoutsbot/venv/lib/python3.5/site-packages/telepot/aio/api.py", line 100, in _parse
    raise exception.TelegramError(description, error_code, data)
telepot.exception.TelegramError: ('Bad Request: can\'t parse message text: Unsupported start tag "br/" at byte offset 23', 400, {'description': 'Bad Request: can\'t parse message text: Unsupported start tag "br/" at byte offset 23', 'ok': False, 'error_code': 400})
```